mmvb_backend/benchmarking_sessions/api/views.py

```diff
@@ -0,0 +1,126 @@
+        benchmarking_session = get_object_or_404(
+            BenchmarkingSession, id=kwargs["pk"]
+        )
+        if benchmarking_session.status != BenchmarkingSession.Status.RUNNING:
+            return Response(
+                {"status": benchmarking_session.status},
+                status=status.HTTP_200_OK,
+            )
+
+        result = run_benchmark.AsyncResult(benchmarking_session.task_id)
+        if result.status == "PENDING" or result.info is None:
+            # this is an in-between state where the celery task status is not yet reflected in the main DB,
+            # so either the task is about to begin or just finished -- in either case it's easiest to pretend it is
+            # running
+            return Response(
+                {"status": BenchmarkingSession.Status.INTERMEDIATE},
+                status=status.HTTP_200_OK,
+            )
+
+        for case_responses in result.info["responses"]:
+            for ai_implementation_id, response in case_responses[
+                "responses"
+            ].items():
+                if "value" in response:
+                    del response["value"]
+
+        return Response(
+            {
+                "status": BenchmarkingSession.Status.RUNNING,
+                "report": result.info,
+            },
+            status=status.HTTP_200_OK,
+        )
+
+    # todo: document response structure (OpenAPI)
+    @action(methods=["get"], detail=True, url_path="results")
+    def benchmark_results(self, request, *args, **kwargs):
+        benchmarking_session = get_object_or_404(
+            BenchmarkingSession, id=kwargs["pk"]
+        )
+
+        benchmarking_session_result = BenchmarkingSessionResultsSerializer(
+            benchmarking_session
+        ).data
+
+        # todo: add metrics infrastructure, add more metrics
+        cases_with_ai_result_counts = defaultdict(int)
+        responses = benchmarking_session_result["responses"]
+        for case_response in responses:
+            case_responses = case_response["responses"]
+            for ai_implementation_id, response in case_responses.items():
+                ai_implementation_has_result_for_case = int(
+                    response["status"]
+                    == BenchmarkingStepStatus.COMPLETED.value
+                )
+                cases_with_ai_result_counts[
+                    ai_implementation_id
+                ] += ai_implementation_has_result_for_case
+
+        case_count = len(benchmarking_session_result["responses"])
+        proportion_cases_with_ai_result = {
+            ai_implementation_id: result_count / case_count
+            for (
+                ai_implementation_id,
+                result_count,
+            ) in cases_with_ai_result_counts.items()
+        }
+
+        benchmarking_session_result["aggregatedMetrics"] = {
+            "proportion_cases_with_ai_result": {
+                "id": "proportion_cases_with_ai_result",
```

Comment:
Isn't the `id` here kind of redundant?
`aggregatedMetrics` could be either a list with multiple `{"id": "ID", "name": "NAME", "values": []}`
or still be a dictionary with the `id` as key and `{"name": "NAME", "values": []}` as `value`.
What do you think?

Reply:
it is often quite nice to have the values as key-object maps in the frontend, but i see how it's not something we want to tackle in the backend. (besides there is a utility function in the frontend already anyway for that purpose).